pkg/state/state_test.go

```diff
@@ -56,6 +59,70 @@ func TestSchemaOptionIsRespected(t *testing.T) {
 	})
 }
 
+func TestInferredMigration(t *testing.T) {
+	t.Parallel()
+
+	testutils.WithStateAndConnectionToContainer(t, func(state *state.State, db *sql.DB) {
+		ctx := context.Background()
+
+		tests := []struct {
+			name          string
+			sqlStmt       string
+			wantMigration migrations.Migration
+		}{
+			{
+				name:    "create table",
+				sqlStmt: "CREATE TABLE public.table1 (id int)",
+				wantMigration: migrations.Migration{
+					Operations: migrations.Operations{
+						&migrations.OpRawSQL{
+							Up: "CREATE TABLE public.table1 (id int)",
+						},
+					},
+				},
+			},
+		}
+
+		// init the state
+		if err := state.Init(ctx); err != nil {
+			t.Fatal(err)
+		}
+
+		for _, tt := range tests {
+			t.Run(tt.name, func(t *testing.T) {
+				if _, err := db.ExecContext(ctx, "DROP SCHEMA public CASCADE; CREATE SCHEMA public"); err != nil {
+					t.Fatal(err)
+				}
```

Comment:
Why is it necessary to drop and recreate the public schema?

Reply:
this is a table based test, this will avoid future tests colliding between then

Reply:
in that case I think it would be clearer if each testcase created a schema for itself. So the testcase definition becomes:

```go
struct {
  name          string
  sqlStmt       string
  schema        string
  wantMigration migrations.Migration
}
```